tests/unit/test_firehose.py

```diff
@@ -1,27 +1,59 @@
 import unittest
+import json
 
 from localstack.services.firehose import firehose_api
 from localstack.utils.common import short_uid
+from localstack.services.infra import start_proxy
+from localstack.services.generic_proxy import ProxyListener
 
-TEST_STREAM_NAME = 'firehose_test_' + short_uid()
+TEST_STREAM_NAME_1 = 'firehose_test_' + short_uid()
+TEST_STREAM_NAME_2 = 'firehose_test_' + short_uid()
 TEST_TAG_1 = {'Key': 'MyTag', 'Value': 'TestValue'}
 TEST_TAG_2 = {'Key': 'AnotherTag', 'Value': 'AnotherValue'}
 TEST_TAGS = [TEST_TAG_1, TEST_TAG_2]
+HTTP_DESTINATION = {'EndpointConfiguration': {'Url': 'http://localhost:5000'}}
+HTTP_DESTINATION_UPDATE = {'EndpointConfiguration': {'Name': 'test_update'}}
 
 
 class FirehoseApiTest(unittest.TestCase):
 
     def setUp(self):
-        firehose_api.create_stream(TEST_STREAM_NAME, tags=TEST_TAGS)
+        firehose_api.create_stream(TEST_STREAM_NAME_1, tags=TEST_TAGS)
+        firehose_api.create_stream(TEST_STREAM_NAME_2, tags=TEST_TAGS, http_destination=HTTP_DESTINATION)
 
     def tearDown(self):
-        firehose_api.delete_stream(TEST_STREAM_NAME)
+        firehose_api.delete_stream(TEST_STREAM_NAME_1)
+        firehose_api.delete_stream(TEST_STREAM_NAME_2)
 
     def test_delivery_stream_tags(self):
-        result = firehose_api.get_delivery_stream_tags(TEST_STREAM_NAME)
+        result = firehose_api.get_delivery_stream_tags(TEST_STREAM_NAME_1)
         self.assertEquals(TEST_TAGS, result['Tags'])
-        result = firehose_api.get_delivery_stream_tags(TEST_STREAM_NAME, exclusive_start_tag_key='MyTag')
+        result = firehose_api.get_delivery_stream_tags(TEST_STREAM_NAME_1, exclusive_start_tag_key='MyTag')
         self.assertEquals([TEST_TAG_2], result['Tags'])
-        result = firehose_api.get_delivery_stream_tags(TEST_STREAM_NAME, limit=1)
+        result = firehose_api.get_delivery_stream_tags(TEST_STREAM_NAME_1, limit=1)
         self.assertEquals([TEST_TAG_1], result['Tags'])
         self.assertEquals(True, result['HasMore'])
+
+    def test_create_stream_with_HttpEndpointDestination(self):
```

Comment:
nitpick: Let's better try to use all lowercase method names consistently across the codebase.
```
def test_create_stream_with_http_destination(self):
    ...
```
(same for lines 44 and 53)